server/testdriver/core/src/main/java/org/infinispan/server/test/core/ContainerInfinispanServerDriver.java

```diff
@@ -296,20 +311,17 @@ private static Path cleanServerDirectory(Path serverOutputPath) {
       return serverOutputPath;
    }
 
-   private GenericContainer<?> createContainer(int i, Consumer<OutputFrame>... logConsumers) {
+   private GenericContainer<?> createContainer(Consumer<OutputFrame>... logConsumers) {
 
       boolean volumesRequired = Boolean.parseBoolean(configuration.properties().getProperty(INFINISPAN_TEST_SERVER_CONTAINER_VOLUME_REQUIRED));
-      if (volumesRequired && this.volumes[i] == null) {
-         String volumeName = Util.threadLocalRandomUUID().toString();
-         DOCKER_CLIENT.createVolumeCmd().withName(volumeName).exec();
-         this.volumes[i] = volumeName;
```

Comment:
I think the idea of storing the volume names was so that we can cleanup and remove the volumes once the testsuite completes. Looking at the diff though, it seems this is missing. Can you try adding this logic? I don't think it's a good thing if our testsuite is continually creating new volumes and not removing them after use.

Reply:
The `volumes` array is never used outside of this though, so I never saw how it could be used to cleanup. The `containers` list/array however is stopped at the end of the test. In my docker environment after running the upgrade everything is cleaned up as I would think. Is there something else I am missing?

Reply:
But I am missing a chance to ensure the volume mount is reused on a given offset with this change though.

Reply:
Will revert, I was able to change the parts of code that this simplified to be different.

Reply:
> The `volumes` array is never used outside of this though, so I never saw how it could be used to cleanup. Maybe the code was actually missing before as well?

Yes the cleanup was missing before, I was asking if you could add the cleanup logic. For example, we could have a `InfinispanServerDriver#close` method which is called only at the end of the test. That way we can still stop/kill/restart the server during the test and only remove the volume when we know for sure that the test no longer relies on it.

For debugging purposes, we could even have a property in the `#close` implementation that preserves the volume.